changelog/7616.improvement.md

```diff
@@ -0,0 +1,6 @@
+Constrain similarity values to an approximate range in `DotProductLoss` by applying sigmoid over them during training.
+
+This affects the default behaviour of the loss function(`loss_type=softmax`) inside machine learning(ML) components - `DIETClassifier`, `ResponseSelector` and `TEDPolicy`.
```

Comment:
I'd suggest to rename `softmax` to cross-entropy loss

Reply:
Deprecated the value `softmax` for `loss_type` in favour of `cross_entropy` with a deprecation warning.